quantities/numerics/Polynomial.cs

```diff
@@ -0,0 +1,78 @@
+    public static Polynomial LinearUp(in Double scaling, in Double offset) => new LinearUpPoly(in scaling, in offset);
+    public static Polynomial LinearDown(in Double scaling, in Double offset) => new LinearDownPoly(in scaling, in offset);
+    public static Polynomial Fractional(in Double nominator, in Double denominator) => new FractionalPoly(in nominator, in denominator);
+    public static Polynomial Full(in Double nominator, in Double denominator, in Double offset) => new FullPoly(in nominator, in denominator, in offset);
+
+    private sealed class NoOpPoly : Polynomial
+    {
+        public override Double Evaluate(in Double value) => value;
+        public override Double Inverse(in Double value) => value;
+    }
+    private sealed class ShiftPoly : Polynomial
+    {
+        private readonly Double offset;
+        public ShiftPoly(in Double offset) => this.offset = offset;
+        public override Double Evaluate(in Double value) => value + this.offset;
+        public override Double Inverse(in Double value) => value - this.offset;
+    }
+    private sealed class ScaleUpPoly : Polynomial
+    {
+        private readonly Double scaling;
+        public ScaleUpPoly(in Double scaling) => this.scaling = scaling;
+        public override Double Evaluate(in Double value) => this.scaling * value;
+        public override Double Inverse(in Double value) => value / this.scaling;
+    }
+    private sealed class ScaleDownPoly : Polynomial
+    {
+        private readonly Double scaling;
+        public ScaleDownPoly(in Double scaling) => this.scaling = scaling;
+        public override Double Evaluate(in Double value) => value / this.scaling;
+        public override Double Inverse(in Double value) => this.scaling * value;
+    }
+    private sealed class LinearUpPoly : Polynomial
+    {
+        private readonly Double scaling, offset;
+        public LinearUpPoly(in Double scaling, in Double offset) => (this.scaling, this.offset) = (scaling, offset);
+        public override Double Evaluate(in Double value) => this.scaling * value + this.offset;
+        public override Double Inverse(in Double value) => (value - this.offset) / this.scaling;
+    }
+    private sealed class LinearDownPoly : Polynomial
+    {
+        private readonly Double scaling, offset;
+        public LinearDownPoly(in Double scaling, in Double offset) => (this.scaling, this.offset) = (scaling, offset);
+        public override Double Evaluate(in Double value) => value / this.scaling + this.offset;
+        public override Double Inverse(in Double value) => this.scaling * (value - this.offset);
+    }
+
+    private sealed class FractionalPoly : Polynomial
+    {
+        private readonly Double nominator, denominator;
+        public FractionalPoly(in Double nominator, in Double denominator) => (this.nominator, this.denominator) = (nominator, denominator);
+        public override Double Evaluate(in Double value) => this.nominator * value / this.denominator;
```

Comment:
If we want to avoid a precision loss, shouldn't the fraction be also expressed as `{nominator, denominator}` so that the `Evaluate` method becomes?

```C#
public override Double Evaluate(in Double value) => this.nominator * fraction.Nominator / this.denominator * fraction.Denominator;
```

Reply:
I'm not quite sure what you mean here, as the `FractionalPoly` is already represented by a fraction. (By the `nominator` and `denominator`fields.)

If you mean representing the value we use for computation, I've already tried that using Dekker's "Double-Double" arithmetic. See commit No. 7d21722 where I had experimented with a Dekker type..

Reply:
It was a misunderstanding. I thought that both would be fractional :)